### PR TITLE
[RW-9955][risk=no] Made equivalents of TestMockFactory that do not require repositories.

### PR DESCRIPTION
**Pull request by evrii**
**Description**

I am hoping to not use a database in the contract test. A few of the TestMockFactory functions utilize daos by default. This PR splits some of those functions between creation and saving them.

Ran unit tests locally and they passed.

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md)
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md)
- [ ] If this PR is intended to complete a JIRA story, I have checked that all AC are met for that story
- [x] I have manually tested this change and my testing process is described above
- [ ] This PR includes appropriate automated tests, and I have documented any behavior that cannot be tested with code
- [ ] If this fixes a bug, ensure the steps to reproduce are in the Jira ticket or provided above.
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented the impacts in the description
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this includes an API change, I have run the relevant E2E tests locally because API changes are not covered by our PR checks

### CODE REVIEW

File: api/src/test/java/org/pmiops/workbench/utils/TestMockFactory.java [hunk 1]
```diff
@@ -256,29 +256,34 @@ public static DbWorkspace createDbWorkspaceStub(Workspace workspace, long worksp
...
+        .setAccessTierId(1)
+        .setShortName(AccessTierService.REGISTERED_TIER_SHORT_NAME)
+        .setDisplayName("Registered Tier")
+        .setAuthDomainName("Registered Tier Auth Domain")
+        .setAuthDomainGroupEmail("rt-users@fake-research-aou.org")
+        .setServicePerimeter("registered/tier/perimeter")
+        .setEnableUserWorkflows(false);
+  }
+
   public static DbAccessTier createRegisteredTierForTests(AccessTierDao accessTierDao) {
```
[the remark this answers is not in the document]
Comment (evrii) [reply api/src/test/java/org/pmiops/workbench/utils/TestMockFactory.java]: I replaced old calls to these functions to use the new functions and dao saving within the test. Please let me know if this looks good to you.

File: api/src/test/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceTest.java [hunk 2]
```diff
@@ -169,7 +170,9 @@ DbUser user() {
   public void setUp() {
     currentUser = new DbUser();
 
-    cdrVersion = TestMockFactory.createDefaultCdrVersion(cdrVersionDao, accessTierDao);
+    cdrVersion = createDefaultCdrVersion();
+    accessTierDao.save(cdrVersion.getAccessTier());
+    cdrVersionDao.save(cdrVersion);
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceTest.java:175]: `cdrVersion = `

File: api/src/test/java/org/pmiops/workbench/workspaces/WorkspaceDaoTest.java [hunk 3]
```diff
@@ -132,7 +133,10 @@ private DbWorkspace createWorkspace() {
     workspace.setWorkspaceNamespace(WORKSPACE_NAMESPACE);
     workspace.setGoogleProject(GOOGLE_PROJECT);
     workspace.setWorkspaceActiveStatusEnum(WorkspaceActiveStatus.ACTIVE);
-    workspace.setCdrVersion(TestMockFactory.createDefaultCdrVersion(cdrVersionDao, accessTierDao));
+    DbCdrVersion dbCdrVersion = createDefaultCdrVersion();
+    accessTierDao.save(dbCdrVersion.getAccessTier());
+    cdrVersionDao.save(dbCdrVersion);
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/workspaces/WorkspaceDaoTest.java:138]: `dbCdrVersion =`

File: api/src/test/java/org/pmiops/workbench/workspaces/WorkspaceServiceTest.java [hunk 4]
```diff
@@ -570,8 +571,9 @@ public void userWithoutCtTierAccessCTWorkspace() {
             "Controlled Tier Workspace",
             DEFAULT_WORKSPACE_NAMESPACE,
             WorkspaceActiveStatus.ACTIVE);
-    DbCdrVersion dbCdrVersion =
-        TestMockFactory.createControlledTierCdrVersion(cdrVersionDao, accessTierDao, 1);
+    DbCdrVersion dbCdrVersion = createControlledTierCdrVersion(1);
+    accessTierDao.save(dbCdrVersion.getAccessTier());
+    cdrVersionDao.save(dbCdrVersion);
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/workspaces/WorkspaceServiceTest.java:576]: `dbCdrVersion =`

File: api/src/test/java/org/pmiops/workbench/notebooks/NotebooksServiceTest.java [hunk 5]
```diff
@@ -116,8 +116,10 @@ public void setup() {
 
     // workspaceDao is a mock, so we don't need to save the workspace
     dbWorkspace = new DbWorkspace();
-    dbWorkspace.setCdrVersion(
-        TestMockFactory.createDefaultCdrVersion(cdrVersionDao, accessTierDao));
+    DbCdrVersion cdrVersion = createDefaultCdrVersion();
+    accessTierDao.save(cdrVersion.getAccessTier());
+    cdrVersionDao.save(cdrVersion);
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/notebooks/NotebooksServiceTest.java:121]: `cdrVersion =`

File: api/src/test/java/org/pmiops/workbench/api/UserMetricsControllerTest.java [hunk 6]
```diff
@@ -117,8 +117,10 @@ static class Configuration {}
 
   @BeforeEach
   public void setUp() {
-    final DbCdrVersion dbCdrVersion =
-        TestMockFactory.createDefaultCdrVersion(cdrVersionDao, accessTierDao);
+    final DbCdrVersion dbCdrVersion = createDefaultCdrVersion();
+
+    accessTierDao.save(dbCdrVersion.getAccessTier());
+    cdrVersionDao.save(dbCdrVersion);
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/api/UserMetricsControllerTest.java:123]: `dbCdrVersion =`

File: api/src/test/java/org/pmiops/workbench/api/OfflineRuntimeControllerTest.java [hunk 7]
```diff
@@ -111,7 +112,9 @@ public WorkbenchConfig workbenchConfig() {
   @BeforeEach
   public void setUp() {
     runtimeProjectIdIndex = 0;
-    DbCdrVersion cdrVersion = TestMockFactory.createDefaultCdrVersion(cdrVersionDao, accessTierDao);
+    DbCdrVersion cdrVersion = createDefaultCdrVersion();
+    accessTierDao.save(cdrVersion.getAccessTier());
+    cdrVersionDao.save(cdrVersion);
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/api/OfflineRuntimeControllerTest.java:117]: `dbCdrVersion =`

File: api/src/test/java/org/pmiops/workbench/api/DataDictionaryTest.java [hunk 8]
```diff
@@ -87,7 +87,9 @@ static class Configuration {}
 
   @BeforeEach
   public void setUp() {
-    cdrVersion = TestMockFactory.createDefaultCdrVersion(cdrVersionDao, accessTierDao);
+    cdrVersion = createDefaultCdrVersion();
+    accessTierDao.save(cdrVersion.getAccessTier());
+    cdrVersionDao.save(cdrVersion);
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/api/DataDictionaryTest.java:92]: `dbCdrVersion =`

File: api/src/test/java/org/pmiops/workbench/api/ConceptSetsControllerTest.java [hunk 9]
```diff
@@ -295,7 +296,9 @@ public void setUp() throws Exception {
     user = userDao.save(user);
     currentUser = user;
 
-    DbCdrVersion cdrVersion = TestMockFactory.createDefaultCdrVersion(cdrVersionDao, accessTierDao);
+    DbCdrVersion cdrVersion = createDefaultCdrVersion();
+    accessTierDao.save(cdrVersion.getAccessTier());
+    cdrVersionDao.save(cdrVersion);
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/api/ConceptSetsControllerTest.java:301]: `dbCdrVersion =`

File: api/src/test/java/org/pmiops/workbench/api/CohortsControllerTest.java [hunk 10]
```diff
@@ -261,7 +262,8 @@ public void setUp() {
     user = userDao.save(user);
     currentUser = user;
 
-    cdrVersion = TestMockFactory.createDefaultCdrVersion(cdrVersionDao, accessTierDao);
+    cdrVersion = createDefaultCdrVersion();
+    accessTierDao.save(cdrVersion.getAccessTier());
     cdrVersion.setName(CDR_VERSION_NAME);
     cdrVersionDao.save(cdrVersion);
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/api/CohortsControllerTest.java:268]: This existed in the original, but we should correct it: `cdrVersion =`

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java [hunk 11]
```diff
@@ -336,7 +337,8 @@ public void setUp() {
     user.setDisabled(false);
     currentUser = userDao.save(user);
 
-    cdrVersion = TestMockFactory.createDefaultCdrVersion(cdrVersionDao, accessTierDao);
+    cdrVersion = createDefaultCdrVersion();
+    accessTierDao.save(cdrVersion.getAccessTier());
     cdrVersion.setName(CDR_VERSION_NAME);
     cdrVersionDao.save(cdrVersion);
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:343]: `cdrVersion =`

**Review by jmthibault79 — APPROVED** [review]
Looks good.  Thanks for rolling with it when I greatly expanded the PR :)

A few cases where we should use the pattern `foo = dbSave(foo)` to keep Hibernate happy

**Pull request merged**